src/uct/rocm/copy/rocm_copy_ep.c

```diff
@@ -69,52 +100,115 @@ ucs_status_t uct_rocm_copy_ep_zcopy(uct_ep_h tl_ep,
     uct_rocm_copy_key_t *rocm_copy_key = (uct_rocm_copy_key_t *) rkey;
 
     hsa_status_t status;
-    hsa_agent_t agent;
-    void *src_addr, *dst_addr;
-    void *host_ptr, *dev_ptr, *mapped_ptr;
-    size_t offset;
+    void *src_addr, *dst_addr, *dev_address;
+    void *remote_addr_mod = NULL, *iov_buffer_mod = NULL;
+    hsa_agent_t agent, tmp_agent;
+    hsa_device_type_t dev_type;
+    hsa_amd_pointer_type_t remote_addr_mem_type, iov_buffer_mem_type;
+    bool remote_addr_is_host = false, iov_buffer_is_host = false;
```

Comment:
initialized variables should be first
use 0/1 instead of true/false

Reply:
done